### PR TITLE
CI-98680 fixed security issues

### PR DESCRIPTION
**Pull request by Lambert03**
Fixed all dependence issues via npm/yarn.
Authorization-code-flow:
Screenshot 2023-05-29 at 10 53 02 am
Device flow:
Screenshot 2023-05-29 at 11 06 59 am
Screenshot 2023-05-29 at 10 40 51 am
Implicity flow:
Screenshot 2023-05-29 at 10 14 08 am
ROPC flow:
Screenshot 2023-05-29 at 10 51 48 am

### CODE REVIEW

**Comment by Lambert03** [comment]
Found errors in implicity react app, will get it fix soon

**Comment by Lambert03** [comment]
> Found errors in implicity react app(react-script v5.0.1), it also get failed(react-script v4.x.x) in downloaded app. will get it fix soon

Downgrade react-script to v4 and fixed dependence issues

File: dist/AuthenticatorContext.js
Comment (trumanbutter) [inline dist/AuthenticatorContext.js:1]: @Lambert03 why are we committing empty file?

File: dist/config.js
Comment (trumanbutter) [inline dist/config.js:1]: @Lambert03 why are we committing empty file?

File: dist/errors/Errors.js
Comment (trumanbutter) [inline dist/errors/Errors.js:1]: @Lambert03 why are we committing empty file?

File: dist/helpers/StorageHandler.js
Comment (trumanbutter) [inline dist/helpers/StorageHandler.js:1]: @Lambert03 why are we committing empty file?

File: dist/helpers/apiRequest.js
Comment (trumanbutter) [inline dist/helpers/apiRequest.js:1]: @Lambert03 why are we committing empty file?

File: dist/helpers/utils.js
Comment (trumanbutter) [inline dist/helpers/utils.js:1]: @Lambert03 why are we committing empty file?

File: dist/index.js
Comment (trumanbutter) [inline dist/index.js:1]: @Lambert03 why are we committing empty file?

File: samples/implicit-flow/react/package.json [hunk 1]
```diff
@@ -7,19 +7,27 @@
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
     "bootstrap": "^5.0.1",
+    "child_process": "^1.0.2",
+    "fs": "^0.0.1-security",
+    "http": "^0.0.1-security",
+    "https": "^1.0.0",
+    "https-browserify": "^1.0.0",
```
Comment (trumanbutter) [inline samples/implicit-flow/react/package.json:14]: Why are we adding these new dependencies?

File: samples/implicit-flow/react/package.json [hunk 2]
```diff
@@ -7,19 +7,27 @@
...
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.3",
     "reactstrap": "^8.9.0",
-    "web-vitals": "^1.0.1"
+    "stream-http": "^3.2.0",
+    "url": "^0.11.0",
+    "web-vitals": "^1.0.1",
+    "xmlhttprequest": "^1.8.0"
```
Comment (trumanbutter) [inline samples/implicit-flow/react/package.json:24]: Why are we adding new dependencies?

File: samples/implicit-flow/react/package.json [hunk 3]
```diff
@@ -7,19 +7,27 @@
...
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "react-scripts --openssl-legacy-provider start",
+    "build": "react-scripts --openssl-legacy-provider build",
+    "test": "react-scripts --openssl-legacy-provider test",
+    "eject": "react-scripts --openssl-legacy-provider eject"
```
Comment (trumanbutter) [inline samples/implicit-flow/react/package.json:30]: @Lambert03 what does the `--openssl-legacy-provider` do when running any of these scripts?

File: samples/implicit-flow/react/yarn.lock
Comment (trumanbutter) [inline samples/implicit-flow/react/yarn.lock:1]: I don't think we should be using yarn as well as npm as an optional build tool as it's known to cause issues.
Let's remove this and just use npm CLI.

File: yarn.lock
Comment (trumanbutter) [inline yarn.lock:1]: @Lambert03 I don't think we should be using yarn as well as npm as an optional build tool as it's known to cause issues.
Let's remove this and just use npm CLI.

**Review by trumanbutter — COMMENTED** [review]
@Lambert03 A few questions about this PR before we can merge.

File: samples/implicit-flow/react/package.json (hunk 2 above)
Comment (Lambert03) [reply samples/implicit-flow/react/package.json:24]: I will try to remove it and it should be added by `npm audit fix`

**Comment by trumanbutter** [comment]
Closing this PR out as all outstanding issues have now been resolved.

**Pull request closed without merging**